MoviesExample/app/src/main/java/com/novoda/demo/movies/MovieService.java

```diff
@@ -1,107 +1,64 @@
 package com.novoda.demo.movies;
 
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.util.Log;
+
 import com.novoda.demo.movies.api.MoviesApi;
 import com.novoda.demo.movies.api.MoviesResponse;
 import com.novoda.demo.movies.api.VideosResponse;
 import com.novoda.demo.movies.model.Movie;
 import com.novoda.demo.movies.model.Video;
 
-import java.util.ArrayList;
 import java.util.List;
 
 import retrofit2.Call;
+import retrofit2.Callback;
 import retrofit2.Response;
-import rx.Observable;
-import rx.Observer;
-import rx.functions.Func1;
-import rx.schedulers.Schedulers;
 
 public class MovieService {
 
     private final MoviesApi api;
-    private MoviesSate moviesSate = new MoviesSate(new ArrayList<Movie>(), 1);
-
-    private Callback callback;
+    private final MutableLiveData<List<Movie>> moviesLiveData = new MutableLiveData<>();
+    private final MutableLiveData<List<Video>> videosLiveData = new MutableLiveData<>();
 
     public MovieService(MoviesApi api) {
         this.api = api;
     }
 
-    public void subscribe(Callback callback) {
-        this.callback = callback;
-        callback.onNewData(moviesSate);
-        if (moviesSate.isEmpty()) {
-            loadMore();
-        }
-    }
-
-    public void unsubscribe(Callback callback) {
-        this.callback = null;
-    }
-
-    public void loadMore() {
-        api.topRated(moviesSate.pageNumber()).enqueue(new retrofit2.Callback<MoviesResponse>() {
+    public LiveData<List<Movie>> loadMore(int page) {
+        api.topRated(page).enqueue(new Callback<MoviesResponse>() {
             @Override
             public void onResponse(Call<MoviesResponse> call, Response<MoviesResponse> response) {
                 if (response == null || response.body() == null || response.body().results == null) {
                     return;
                 }
-                List<Movie> movies = moviesSate.movies();
-                movies.addAll(response.body().results);
-                moviesSate = new MoviesSate(movies, moviesSate.pageNumber() + 1);
-                callback.onNewData(moviesSate);
+                moviesLiveData.postValue(response.body().results);
```

Comment:
`set` should only be used on the main thread, post on other threads: https://developer.android.com/topic/libraries/architecture/livedata#update_livedata_objects

Reply:
This is the main thread no?

Reply:
Yes, @ataulm is right: `enqueue` calls back into the main thread so in this case you can use `moviesLiveData.setValue()` too

Reply:
are there any drawbacks to using post? e.g. for simplicity, could one ONLY use post?

Reply:
AFAIK there's no difference for the observer. From the documentation:
>  * `setValue`:  Sets the value. If there are active observers, the value will be dispatched to them. This method must be called from the main thread
> * `postValue`: Posts a task to a main thread to set the given value. If you called this method multiple times before a main thread executed a posted task, only the last value would be dispatched.

So both can be called from the main thread but only `postValue` can be called from a background thread. That "if" could be the only real use case difference.

Also from the docs: 
> If you have a following code executed in the main thread:
> ```
> liveData.postValue("a");
> liveData.setValue("b");
> ```
> The value "b" would be set at first and later the main thread would override it with the value "a".

Something to have in mind if you use both `setValue` and `postValue`

Reply:
in this case do you think it's worth using `setValue` since it's in the main thread?

If we use these as examples, I'd rather someone defaults to `setValue` and then is _forced_ to read up on `postValue` because `setValue` is impossible to use on a background thread, rather than using `postValue` and getting caught out by the last example you gave.